Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 30, in <module>
    api = tradeapi.REST(ALPACA_API_KEY, ALPACA_API_SECRET, ALPACA_BASE_URL, api_version='v2')
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 177, in __init__
    self._key_id, self._secret_key, self._oauth = get_credentials(
                                                  ^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/common.py", line 84, in get_credentials
    raise ValueError('Key ID must be given to access Alpaca trade API',
ValueError: ('Key ID must be given to access Alpaca trade API', ' (env: APCA_API_KEY_ID)')
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 12, in <module>
    from config import (
ImportError: cannot import name 'ALPACA_API_KEY' from 'config' (/Users/Cook/Documents/VS Code/Moving Average Calc/config.py)
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 42, in <module>
    api = tradeapi.REST(ALPACA_API_KEY, ALPACA_API_SECRET, ALPACA_BASE_URL, api_version='v2')
                                                           ^^^^^^^^^^^^^^^
NameError: name 'ALPACA_BASE_URL' is not defined
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 42, in <module>
    api = tradeapi.REST(ALPACA_API_KEY, ALPACA_API_SECRET, ALPACA_BASE_URL, api_version='v2')
                                                           ^^^^^^^^^^^^^^^
NameError: name 'ALPACA_BASE_URL' is not defined
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 160, in <module>
    datetime.time.sleep(60)  # Wait for 60 seconds
    ^^^^^^^^^^^^^^^^^^^
AttributeError: 'method_descriptor' object has no attribute 'sleep'
2024-03-04 08:47:00,386 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-03-04 08:47:00,841 - INFO - Email sent successfully
2024-03-04 08:47:00,841 - INFO - Could not retrieve stock data.
2024-03-04 08:50:44,588 - INFO - Email sent successfully
2024-03-05 08:47:00,356 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-03-05 08:47:00,822 - INFO - Email sent successfully
2024-03-05 08:47:00,823 - INFO - Could not retrieve stock data.
2024-03-05 08:47:00,825 - INFO - Email sent successfully
2024-03-05 08:47:00,829 - INFO - Email sent successfully
2024-03-06 08:47:00,410 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-03-06 08:47:00,734 - INFO - Email sent successfully
2024-03-06 08:47:00,735 - INFO - Email sent successfully
2024-03-06 08:47:00,737 - INFO - Email sent successfully
2024-03-06 08:47:00,737 - INFO - Could not retrieve stock data.
2024-03-07 08:47:00,514 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-03-07 08:47:00,919 - INFO - Email sent successfully
2024-03-07 08:47:00,924 - INFO - Email sent successfully
2024-03-07 08:47:00,928 - INFO - Email sent successfully
2024-03-07 08:47:00,928 - INFO - Could not retrieve stock data.
2024-03-08 08:47:00,350 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-03-08 08:47:00,714 - INFO - Email sent successfully
2024-03-08 08:47:00,714 - INFO - Could not retrieve stock data.
2024-03-08 08:47:00,718 - INFO - Email sent successfully
2024-03-08 08:47:00,718 - INFO - Email sent successfully
2024-03-11 08:47:00,416 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-03-11 08:47:00,761 - INFO - Email sent successfully
2024-03-11 08:47:00,762 - INFO - Could not retrieve stock data.
2024-03-11 08:47:00,762 - INFO - Email sent successfully
2024-03-11 08:47:00,766 - INFO - Email sent successfully
2024-03-12 08:47:00,327 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-03-12 08:47:00,760 - INFO - Email sent successfully
2024-03-12 08:47:00,761 - INFO - Could not retrieve stock data.
2024-03-12 08:47:00,765 - INFO - Email sent successfully
2024-03-12 08:47:00,771 - INFO - Email sent successfully
2024-03-13 08:47:00,718 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-03-13 08:47:00,918 - INFO - Email sent successfully
2024-03-13 08:47:00,971 - INFO - Email sent successfully
2024-03-13 08:47:00,971 - INFO - Could not retrieve stock data.
2024-03-13 08:47:01,093 - INFO - Email sent successfully
2024-03-14 08:47:00,426 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-03-14 08:47:00,749 - INFO - Email sent successfully
2024-03-14 08:47:00,749 - INFO - Could not retrieve stock data.
2024-03-14 08:47:00,756 - INFO - Email sent successfully
2024-03-14 08:47:00,759 - INFO - Email sent successfully
2024-03-15 08:47:00,333 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-03-15 08:47:00,698 - INFO - Email sent successfully
2024-03-15 08:47:00,704 - INFO - Email sent successfully
2024-03-15 08:47:00,704 - INFO - Could not retrieve stock data.
2024-03-15 08:47:00,746 - INFO - Email sent successfully
2024-03-18 08:47:00,326 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-03-18 08:47:00,730 - INFO - Email sent successfully
2024-03-18 08:47:00,732 - INFO - Email sent successfully
2024-03-18 08:47:00,732 - INFO - Could not retrieve stock data.
2024-03-18 08:47:00,732 - INFO - Email sent successfully
2024-03-19 08:47:00,361 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-03-19 08:47:00,818 - INFO - Email sent successfully
2024-03-19 08:47:00,822 - INFO - Email sent successfully
2024-03-19 08:47:00,823 - INFO - Email sent successfully
2024-03-19 08:47:00,823 - INFO - Could not retrieve stock data.
2024-03-20 08:47:00,337 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-03-20 08:47:00,626 - INFO - Email sent successfully
2024-03-20 08:47:00,626 - INFO - Could not retrieve stock data.
2024-03-20 08:47:00,626 - INFO - Email sent successfully
2024-03-20 08:47:00,630 - INFO - Email sent successfully
2024-03-21 08:47:00,328 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-03-21 08:47:00,705 - INFO - Email sent successfully
2024-03-21 08:47:00,707 - INFO - Email sent successfully
2024-03-21 08:47:00,707 - INFO - Could not retrieve stock data.
2024-03-21 08:47:00,709 - INFO - Email sent successfully
2024-03-22 08:47:00,343 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-03-22 08:47:01,706 - INFO - Email sent successfully
2024-03-22 08:47:01,707 - INFO - Could not retrieve stock data.
2024-03-22 08:47:01,715 - INFO - Email sent successfully
2024-03-22 08:47:01,715 - INFO - Email sent successfully
2024-03-25 08:47:03,575 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-03-25 08:47:04,728 - INFO - Email sent successfully
2024-03-25 08:47:04,728 - INFO - Email sent successfully
2024-03-25 08:47:04,728 - INFO - Could not retrieve stock data.
2024-03-25 08:47:04,754 - INFO - Email sent successfully
2024-03-26 08:47:00,339 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-03-26 08:47:00,747 - INFO - Email sent successfully
2024-03-26 08:47:00,747 - INFO - Email sent successfully
2024-03-26 08:47:00,747 - INFO - Could not retrieve stock data.
2024-03-26 08:47:00,800 - INFO - Email sent successfully
2024-03-27 08:47:00,350 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-03-27 08:47:00,709 - INFO - Email sent successfully
2024-03-27 08:47:00,712 - INFO - Email sent successfully
2024-03-27 08:47:00,712 - INFO - Could not retrieve stock data.
2024-03-27 08:47:00,712 - INFO - Email sent successfully
2024-03-28 08:47:00,395 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-03-28 08:47:00,761 - INFO - Email sent successfully
2024-03-28 08:47:00,763 - INFO - Email sent successfully
2024-03-28 08:47:00,763 - INFO - Could not retrieve stock data.
2024-03-28 08:47:00,765 - INFO - Email sent successfully
2024-03-29 08:47:00,348 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-03-29 08:47:00,706 - INFO - Email sent successfully
2024-03-29 08:47:00,707 - INFO - Email sent successfully
2024-03-29 08:47:00,715 - INFO - Email sent successfully
2024-03-29 08:47:00,716 - INFO - Could not retrieve stock data.
2024-04-01 08:47:00,323 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-04-01 08:47:00,722 - INFO - Email sent successfully
2024-04-01 08:47:00,723 - INFO - Could not retrieve stock data.
2024-04-01 08:47:00,726 - INFO - Email sent successfully
2024-04-01 08:47:00,731 - INFO - Email sent successfully
2024-04-02 08:47:00,343 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-04-02 08:47:00,771 - INFO - Email sent successfully
2024-04-02 08:47:00,774 - INFO - Email sent successfully
2024-04-02 08:47:00,788 - INFO - Email sent successfully
2024-04-02 08:47:00,789 - INFO - Could not retrieve stock data.
2024-04-03 08:47:00,489 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-04-03 08:47:00,770 - INFO - Email sent successfully
2024-04-03 08:47:00,777 - INFO - Email sent successfully
2024-04-03 08:47:00,777 - INFO - Could not retrieve stock data.
2024-04-03 08:47:00,798 - INFO - Email sent successfully
2024-04-04 08:47:00,343 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-04-04 08:47:00,739 - INFO - Email sent successfully
2024-04-04 08:47:00,739 - INFO - Email sent successfully
2024-04-04 08:47:00,740 - INFO - Email sent successfully
2024-04-04 08:47:00,740 - INFO - Could not retrieve stock data.
2024-04-05 08:47:00,325 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-04-05 08:47:00,638 - INFO - Email sent successfully
2024-04-05 08:47:00,640 - INFO - Email sent successfully
2024-04-05 08:47:00,641 - INFO - Could not retrieve stock data.
2024-04-05 08:47:00,641 - INFO - Email sent successfully
2024-04-08 08:47:00,320 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-04-08 08:47:00,661 - INFO - Email sent successfully
2024-04-08 08:47:00,661 - INFO - Email sent successfully
2024-04-08 08:47:00,665 - INFO - Email sent successfully
2024-04-08 08:47:00,665 - INFO - Could not retrieve stock data.
2024-04-09 08:47:00,319 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-04-09 08:47:00,676 - INFO - Email sent successfully
2024-04-09 08:47:00,676 - INFO - Could not retrieve stock data.
2024-04-09 08:47:00,676 - INFO - Email sent successfully
2024-04-09 08:47:00,679 - INFO - Email sent successfully
2024-04-10 08:47:00,447 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-04-10 08:47:00,894 - INFO - Email sent successfully
2024-04-10 08:47:00,894 - INFO - Email sent successfully
2024-04-10 08:47:00,911 - INFO - Email sent successfully
2024-04-10 08:47:00,911 - INFO - Could not retrieve stock data.
2024-04-11 08:47:00,385 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-04-11 08:47:00,791 - INFO - Email sent successfully
2024-04-11 08:47:00,794 - INFO - Email sent successfully
2024-04-11 08:47:00,794 - INFO - Could not retrieve stock data.
2024-04-11 08:47:00,794 - INFO - Email sent successfully
2024-04-12 08:47:00,558 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-04-12 08:47:00,933 - INFO - Email sent successfully
2024-04-12 08:47:00,933 - INFO - Could not retrieve stock data.
2024-04-12 08:47:00,935 - INFO - Email sent successfully
2024-04-12 08:47:00,938 - INFO - Email sent successfully
2024-04-15 08:47:00,318 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-04-15 08:47:00,725 - INFO - Email sent successfully
2024-04-15 08:47:00,731 - INFO - Email sent successfully
2024-04-15 08:47:00,731 - INFO - Email sent successfully
2024-04-15 08:47:00,731 - INFO - Could not retrieve stock data.
2024-04-16 08:47:00,335 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-04-16 08:47:01,633 - INFO - Email sent successfully
2024-04-16 08:47:01,633 - INFO - Email sent successfully
2024-04-16 08:47:01,636 - INFO - Email sent successfully
2024-04-16 08:47:01,636 - INFO - Could not retrieve stock data.
2024-04-17 08:47:00,339 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-04-17 08:47:00,662 - INFO - Email sent successfully
2024-04-17 08:47:00,662 - INFO - Could not retrieve stock data.
2024-04-17 08:47:00,667 - INFO - Email sent successfully
2024-04-17 08:47:00,670 - INFO - Email sent successfully
2024-04-18 08:47:00,438 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-04-18 08:47:00,786 - INFO - Email sent successfully
2024-04-18 08:47:00,786 - INFO - Could not retrieve stock data.
2024-04-18 08:47:00,787 - INFO - Email sent successfully
2024-04-18 08:47:00,832 - INFO - Email sent successfully
2024-04-19 08:47:00,347 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-04-19 08:47:00,656 - INFO - Email sent successfully
2024-04-19 08:47:00,660 - INFO - Email sent successfully
2024-04-19 08:47:00,661 - INFO - Email sent successfully
2024-04-19 08:47:00,661 - INFO - Could not retrieve stock data.
2024-04-22 08:47:00,324 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-04-22 08:47:00,735 - INFO - Email sent successfully
2024-04-22 08:47:00,735 - INFO - Could not retrieve stock data.
2024-04-22 08:47:00,737 - INFO - Email sent successfully
2024-04-22 08:47:00,751 - INFO - Email sent successfully
2024-04-23 08:47:00,327 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-04-23 08:47:00,860 - INFO - Email sent successfully
2024-04-23 08:47:00,860 - INFO - Email sent successfully
2024-04-23 08:47:00,861 - INFO - Could not retrieve stock data.
2024-04-23 08:47:00,861 - INFO - Email sent successfully
2024-04-24 08:47:00,317 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-04-24 08:47:00,722 - INFO - Email sent successfully
2024-04-24 08:47:00,722 - INFO - Could not retrieve stock data.
2024-04-24 08:47:00,722 - INFO - Email sent successfully
2024-04-24 08:47:00,737 - INFO - Email sent successfully
2024-04-25 08:47:00,432 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-04-25 08:47:00,839 - INFO - Email sent successfully
2024-04-25 08:47:00,839 - INFO - Could not retrieve stock data.
2024-04-25 08:47:00,839 - INFO - Email sent successfully
2024-04-25 08:47:00,852 - INFO - Email sent successfully
2024-04-26 08:47:00,419 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-04-26 08:47:00,790 - INFO - Email sent successfully
2024-04-26 08:47:00,790 - INFO - Could not retrieve stock data.
2024-04-26 08:47:00,790 - INFO - Email sent successfully
2024-04-26 08:47:00,842 - INFO - Email sent successfully
2024-04-29 08:47:00,335 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-04-29 08:47:01,683 - INFO - Email sent successfully
2024-04-29 08:47:01,683 - INFO - Email sent successfully
2024-04-29 08:47:01,685 - INFO - Email sent successfully
2024-04-29 08:47:01,685 - INFO - Could not retrieve stock data.
2024-04-30 08:47:00,330 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-04-30 08:47:00,743 - INFO - Email sent successfully
2024-04-30 08:47:00,744 - INFO - Email sent successfully
2024-04-30 08:47:00,748 - INFO - Email sent successfully
2024-04-30 08:47:00,748 - INFO - Could not retrieve stock data.
2024-05-01 08:47:00,425 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-05-01 08:47:00,825 - INFO - Email sent successfully
2024-05-01 08:47:00,825 - INFO - Email sent successfully
2024-05-01 08:47:00,826 - INFO - Email sent successfully
2024-05-01 08:47:00,826 - INFO - Could not retrieve stock data.
2024-05-02 08:47:00,318 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-05-02 08:47:00,692 - INFO - Email sent successfully
2024-05-02 08:47:00,692 - INFO - Could not retrieve stock data.
2024-05-02 08:47:00,692 - INFO - Email sent successfully
2024-05-02 08:47:00,711 - INFO - Email sent successfully
2024-05-03 08:47:00,353 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-05-03 08:47:00,834 - INFO - Email sent successfully
2024-05-03 08:47:00,834 - INFO - Email sent successfully
2024-05-03 08:47:00,836 - INFO - Email sent successfully
2024-05-03 08:47:00,836 - INFO - Could not retrieve stock data.
2024-05-06 08:47:00,427 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-05-06 08:47:00,831 - INFO - Email sent successfully
2024-05-06 08:47:00,833 - INFO - Email sent successfully
2024-05-06 08:47:00,834 - INFO - Could not retrieve stock data.
2024-05-06 08:47:00,838 - INFO - Email sent successfully
2024-05-07 08:47:00,354 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-05-07 08:47:00,716 - INFO - Email sent successfully
2024-05-07 08:47:00,718 - INFO - Email sent successfully
2024-05-07 08:47:00,718 - INFO - Email sent successfully
2024-05-07 08:47:00,718 - INFO - Could not retrieve stock data.
2024-05-08 08:47:00,372 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-05-08 08:47:00,722 - INFO - Email sent successfully
2024-05-08 08:47:00,725 - INFO - Email sent successfully
2024-05-08 08:47:00,725 - INFO - Could not retrieve stock data.
2024-05-08 08:47:00,763 - INFO - Email sent successfully
2024-05-09 08:47:00,559 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-05-09 08:47:00,879 - INFO - Email sent successfully
2024-05-09 08:47:00,881 - INFO - Email sent successfully
2024-05-09 08:47:00,883 - INFO - Email sent successfully
2024-05-09 08:47:00,883 - INFO - Could not retrieve stock data.
2024-05-10 08:47:01,370 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-05-10 08:47:01,873 - INFO - Email sent successfully
2024-05-10 08:47:01,873 - INFO - Could not retrieve stock data.
2024-05-10 08:47:01,873 - INFO - Email sent successfully
2024-05-10 08:47:01,875 - INFO - Email sent successfully
2024-05-13 08:47:00,401 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-05-13 08:47:00,644 - INFO - Email sent successfully
2024-05-13 08:47:00,644 - INFO - Email sent successfully
2024-05-13 08:47:00,645 - INFO - Could not retrieve stock data.
2024-05-13 08:47:00,660 - INFO - Email sent successfully
2024-05-14 08:47:00,347 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-05-14 08:47:00,758 - INFO - Email sent successfully
2024-05-14 08:47:00,758 - INFO - Email sent successfully
2024-05-14 08:47:00,761 - INFO - Email sent successfully
2024-05-14 08:47:00,761 - INFO - Could not retrieve stock data.
2024-05-15 08:47:00,328 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-05-15 08:47:00,827 - INFO - Email sent successfully
2024-05-15 08:47:00,827 - INFO - Could not retrieve stock data.
2024-05-15 08:47:00,829 - INFO - Email sent successfully
2024-05-15 08:47:00,829 - INFO - Email sent successfully
2024-05-16 08:47:00,316 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-05-16 08:47:01,470 - INFO - Email sent successfully
2024-05-16 08:47:01,471 - INFO - Email sent successfully
2024-05-16 08:47:01,472 - INFO - Email sent successfully
2024-05-16 08:47:01,472 - INFO - Could not retrieve stock data.
2024-05-17 08:47:00,388 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-05-17 08:47:00,724 - INFO - Email sent successfully
2024-05-17 08:47:00,724 - INFO - Email sent successfully
2024-05-17 08:47:00,725 - INFO - Could not retrieve stock data.
2024-05-17 08:47:00,728 - INFO - Email sent successfully
2024-05-20 08:47:00,521 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-05-20 08:47:00,937 - INFO - Email sent successfully
2024-05-20 08:47:00,938 - INFO - Could not retrieve stock data.
2024-05-20 08:47:00,938 - INFO - Email sent successfully
2024-05-20 08:47:00,942 - INFO - Email sent successfully
2024-05-21 08:47:00,457 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-05-21 08:47:00,814 - INFO - Email sent successfully
2024-05-21 08:47:00,814 - INFO - Email sent successfully
2024-05-21 08:47:00,814 - INFO - Could not retrieve stock data.
2024-05-21 08:47:00,819 - INFO - Email sent successfully
2024-05-22 08:47:00,328 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-05-22 08:47:00,867 - INFO - Email sent successfully
2024-05-22 08:47:00,867 - INFO - Email sent successfully
2024-05-22 08:47:00,867 - INFO - Could not retrieve stock data.
2024-05-22 08:47:00,917 - INFO - Email sent successfully
2024-05-23 08:47:00,355 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-05-23 08:47:00,724 - INFO - Email sent successfully
2024-05-23 08:47:00,724 - INFO - Email sent successfully
2024-05-23 08:47:00,757 - INFO - Email sent successfully
2024-05-23 08:47:00,757 - INFO - Could not retrieve stock data.
2024-05-24 08:47:00,413 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-05-24 08:47:00,725 - INFO - Email sent successfully
2024-05-24 08:47:00,727 - INFO - Email sent successfully
2024-05-24 08:47:00,728 - INFO - Email sent successfully
2024-05-24 08:47:00,728 - INFO - Could not retrieve stock data.
2024-05-27 08:47:01,734 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-05-27 08:47:02,065 - INFO - Email sent successfully
2024-05-27 08:47:02,065 - INFO - Email sent successfully
2024-05-27 08:47:02,110 - INFO - Email sent successfully
2024-05-27 08:47:02,110 - INFO - Could not retrieve stock data.
2024-05-28 08:47:00,358 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-05-28 08:47:00,674 - INFO - Email sent successfully
2024-05-28 08:47:00,680 - INFO - Email sent successfully
2024-05-28 08:47:00,680 - INFO - Could not retrieve stock data.
2024-05-28 08:47:00,704 - INFO - Email sent successfully
2024-05-29 08:47:00,300 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-05-29 08:47:00,653 - INFO - Email sent successfully
2024-05-29 08:47:00,653 - INFO - Email sent successfully
2024-05-29 08:47:00,653 - INFO - Could not retrieve stock data.
2024-05-29 08:47:00,655 - INFO - Email sent successfully
2024-05-30 08:47:00,339 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-05-30 08:47:00,623 - INFO - Email sent successfully
2024-05-30 08:47:00,623 - INFO - Email sent successfully
2024-05-30 08:47:00,627 - INFO - Email sent successfully
2024-05-30 08:47:00,628 - INFO - Could not retrieve stock data.
2024-05-31 08:47:00,540 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-05-31 08:47:00,847 - INFO - Email sent successfully
2024-05-31 08:47:00,847 - INFO - Email sent successfully
2024-05-31 08:47:00,853 - INFO - Email sent successfully
2024-05-31 08:47:00,854 - INFO - Could not retrieve stock data.
2024-06-03 08:47:00,350 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-06-03 08:47:00,800 - INFO - Email sent successfully
2024-06-03 08:47:00,800 - INFO - Email sent successfully
2024-06-03 08:47:00,803 - INFO - Email sent successfully
2024-06-03 08:47:00,803 - INFO - Could not retrieve stock data.
2024-06-04 08:47:00,342 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-06-04 08:47:00,695 - INFO - Email sent successfully
2024-06-04 08:47:00,695 - INFO - Email sent successfully
2024-06-04 08:47:00,743 - INFO - Email sent successfully
2024-06-04 08:47:00,743 - INFO - Could not retrieve stock data.
2024-06-05 08:47:00,334 - ERROR - Exception occurred
Traceback (most recent call last):
  File "/Users/Cook/Documents/VS Code/Moving Average Calc/main.py", line 49, in get_7_day_moving_average
    barset = api.get_bars(symbol, '1D', start=start_date.strftime('%Y-%m-%d'), end=end_date.strftime('%Y-%m-%d'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 735, in get_bars
    bars = list(self.get_bars_iter(symbol,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 718, in get_bars_iter
    for bar in bars:
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 594, in _data_get
    resp = self.data_get(path, data=data, feed=feed,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 274, in data_get
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 222, in _request
    return self._one_request(method, url, opts, retry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 248, in _one_request
    raise_api_error(resp, http_error)
  File "/Users/Cook/Library/Python/3.11/lib/python/site-packages/alpaca_trade_api/rest.py", line 83, in raise_api_error
    raise APIError(error, http_error) from None
alpaca_trade_api.rest.APIError: subscription does not permit querying recent SIP data
2024-06-05 08:47:00,730 - INFO - Email sent successfully
2024-06-05 08:47:00,731 - INFO - Could not retrieve stock data.
2024-06-05 08:47:00,736 - INFO - Email sent successfully
2024-06-05 08:47:00,762 - INFO - Email sent successfully
